### DISCUSSION TITLE
Welcome to Metro-UI-CSS Discussions!

### DISCUSSION THREAD

**Discussion by olton** [olton/Metro-UI-CSS#1692] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by elpepefontana** [comment]
How can i get data out from a $.post or $.get call. I need to set data to pass to another functions.

**Comment by olton** [comment]
https://metroui.org.ua/m4q-ajax.html

**Reply by elpepefontana** [reply]
there there is no information on how to RETURN a value from a success promise.
tried using type: json and contentType: application/json, an a lot of other things
how can i return a value/data from it

**Comment by olton** [comment]
async/await?

**Reply by elpepefontana** [reply]
try using async on false...

function searchComboFieldText(objectName, searchObject, fieldName, id) {
    var url = setURL(searchObject, 'searchById');
    
    var send = {};
    send.id = id;
    
    $.ajax({
        url: url,
        method: "POST",
        data: send,
        dataType: 'json', 
        contentType: 'application/json',
        async: false
    }).then(
        function(response){
            var data = JSON.parse(response);
            if (data.Result !== 'OK') {
                return '';
            } 
            selectValue = data.Records[0][fieldName];
            return selectValue;
        },
        function(xhr) {
            console.log(xhr.status, xhr.statusText);
        }
    );
}
